File: src/classes/api/req_res/update-mi-response.ts
'use strict'

import { Mi } from '@/classes/datas/mi'
import { GkillAPIResponse } from '../gkill-api-response'
import { Kyou } from '@/classes/datas/kyou'

export class UpdateMiResponse extends GkillAPIResponse {

    updated_mi: Mi

    updated_kyou: Kyou | null

    constructor() {
        super()
        this.updated_mi = new Mi()
        this.updated_kyou = null
    }

}

File: src/server/gkill/api/req_res/update_mi_response.go
package req_res

import (
	"github.com/mt3hr/gkill/src/server/gkill/api/message"
	"github.com/mt3hr/gkill/src/server/gkill/dao/reps"
)

type UpdateMiResponse struct {
	Messages []*message.GkillMessage `json:"messages"`

	Errors []*message.GkillError `json:"errors"`

	UpdatedMi *reps.Mi `json:"updated_mi"`

	UpdatedKyou *reps.Kyou `json:"updated_kyou"`
}
